api/src/main/java/com/codeforcommunity/rest/RestFunctions.java

```diff
@@ -83,4 +85,29 @@ public static Country getCountryFromString(String countryName) {
       throw new UnknownCountryException(countryName);
     }
   }
+  /**
+   * Get's a query parameter that may or may not be there as an optional of the desired type.
+   * Attempts to map the query parameter from a string to an instance of the desired type.
+   *
+   * @param ctx routing context to retrieve query param from.
+   * @param name of query param.
+   * @param mapper a function that maps the query param from string to desired type.
+   * @param <T> the desired type.
+   * @return An optional object of the query param as it's desired type.
+   */
+  public static <T> Optional<T> getOptionalQueryParam(
```

Comment:
how did we not have this before?

Reply:
No Idea. I stole it from another of our backend repos